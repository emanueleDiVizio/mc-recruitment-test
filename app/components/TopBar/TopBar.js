import React from 'react'
import { View, Text, StyleSheet } from 'react-native'
import config from '../../config'

const TopBar = ({ title = config.headerTitle  }) => {
  return (
    <View style={styles.container}>
      <Text style={styles.title}>{title}</Text>
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    height: 100,
    backgroundColor: config.colors.main,
    alignItems: 'center',
    justifyContent: 'flex-end',
  },
  title: { color: 'white', marginBottom: 12, fontSize: 17 },
  icon: { flexDirection: 'row' },
})

export default TopBar
