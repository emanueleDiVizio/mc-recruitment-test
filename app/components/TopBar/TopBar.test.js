import React from 'react'
import TopBar from './TopBar'
import { render } from '@testing-library/react-native'

describe('TopBar', () => {
  it('renders correctly when off shift', () => {
    const { baseElement } = render(<TopBar />)

    expect(baseElement).toMatchSnapshot()
  })
})
