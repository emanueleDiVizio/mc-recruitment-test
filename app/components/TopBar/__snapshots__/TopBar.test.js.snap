// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBar renders correctly when off shift 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "backgroundColor": "#00A778",
          "flexDirection": "row",
          "height": 100,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "flex": 1,
            "fontSize": 17,
            "marginBottom": 12,
            "textAlign": "center",
          }
        }
      >
        Wash timer
      </Text>
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 21,
            },
            Object {
              "bottom": 12,
              "flexDirection": "row",
              "position": "absolute",
              "right": 12,
            },
            Object {
              "fontFamily": "FontAwesome5Free-Solid",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {
              "fontWeight": "900",
            },
          ]
        }
      >
        ï€“
      </Text>
    </View>
  </View>
</View>
`;
